lib/yoti/http/request.rb

```diff
@@ -44,7 +59,12 @@ def path
       @path ||= begin
         nonce = SecureRandom.uuid
         timestamp = Time.now.to_i
-        "/profile/#{token}?nonce=#{nonce}&timestamp=#{timestamp}&appId=#{Yoti.configuration.client_sdk_id}"
+
+        "/#{@endpoint}/#{token}"\
+        "?nonce=#{nonce}"\
+        "&timestamp=#{timestamp}"\
+        "&appId=#{Yoti.configuration.client_sdk_id}"\
+        "#{payload.nil? ? '' : '&' + payload_string}"
```

Comment:
I might be wrong, but this is called by uri which is in turn called when creating the request.
If that is the case, you are putting the payload string in the url and that would be wrong.

Reply:
You are right.